### ISSUE TITLE
error in method calc_slope_at_node 

### ISSUE THREAD

**Issue by Amos-Zhao** [landlab/landlab#1608]
i am using `calc_slope_at_node` with method of Horn and the following error occurs. 
I guess the diagonal_neighbors_at_node attribute has been deleted in previous release.
How can i fix it ？ is there any other ways to implement standard ArcGIS slope algorithm?

```shell
Traceback (most recent call last):
  File "D:\code\soil\tetstt.py", line 7, in <module>
    slopes = mg.calc_slope_at_node(elevs=z, method='Horn')
  File "C:\software\python39\lib\site-packages\landlab\grid\raster_gradients.py", line 1854, in calc_slope_at_node
    diags = grid.diagonal_neighbors_at_node.copy()  # LL order
AttributeError: 'RasterModelGrid' object has no attribute 'diagonal_neighbors_at_node'
```

**Comment by gregtucker** [comment]
Here's code that reproduces this:

```
from landlab import RasterModelGrid
import numpy as np

rg = RasterModelGrid((3, 3))
z = rg.add_zeros('topographic__elevation', at='node')
z[:] = np.arange(9)
rg.calc_slope_at_node(elevs=z, method='Horn')
```

I'll post an issue.

**Comment by gregtucker** [comment]
Should be addressed in PR #1609

**Comment by gregtucker** [comment]
I'm going to close this issue, which I think is now resolved.
